crowbar_framework/app/models/api/node.rb

```diff
@@ -50,37 +50,65 @@ def repocheck(options = {})
       end
     end
 
-    protected
+    def initialize(name = nil)
+      @node = NodeObject.find_node_by_name name
+      if @node
+        current_platform = "#{@node[:platform]}-#{@node[:platform_version]}"
+        @upgraded = current_platform == @node[:target_platform]
+      end
+    end
 
-    def node_architectures
-      {}.tap do |ret|
-        NodeObject.all.each do |node|
-          arch = node.architecture
-          ret["os"] ||= []
-          ret["os"].push(arch) unless ret["os"].include?(arch)
+    def pre_upgrade(node)
+      # Migrate out the l3 agents and shut down pacemaker
+      # FIXME: should we make thise code part of crowbar-upgrade-os.sh?
+      node.ssh_cmd("/usr/sbin/crowbar-pre-upgrade.sh")
```

Comment:
... or not. The file is on the remote node, so if it is not there, ssh_cmd should fail itself.